### ISSUE TITLE
f2py doesn't handle bad names in common block

### ISSUE THREAD

**Issue by bobeldering** [numpy/numpy#9228]
numpy version: 1.13.0
component: f2py

Example input:

```
      SUBROUTINE FOO
      DOUBLE PRECISION LONG
      CHARACTER        STRING
      INTEGER          OK
    
      COMMON  /BB/ LONG, STRING, OK
      RETURN
      END

```
Results in signature file:

```
!    -*- f90 -*-
! Note: the context of this file is case sensitive.

python module badname ! in 
    interface  ! in :badname
        subroutine foo ! in :badname:badname.f
            integer :: long
            real :: string
            integer :: ok
            common /bb/ long,string,ok
        end subroutine foo
    end interface 
end python module badname

! This file was auto-generated with f2py (version:2).
! See http://cens.ioc.ee/projects/f2py2e/

```

The implicit rules for LONG and STRING seem to be used to derive their type, with the wrong results.

The following patch fixed it for my use case:

```
*** crackfortran-1.13.0.py      2017-06-08 13:29:55.679736646 +0200
--- crackfortran-fixed.py       2017-06-08 13:30:09.289786804 +0200
*************** def analyzecommon(block):
*** 2013,2019 ****
                  if m.group('dims'):
                      dims = [x.strip()
                              for x in markoutercomma(m.group('dims')).split('@,@')]
!                 n = m.group('name').strip()
                  if n in block['vars']:
                      if 'attrspec' in block['vars'][n]:
                          block['vars'][n]['attrspec'].append(
--- 2013,2019 ----
                  if m.group('dims'):
                      dims = [x.strip()
                              for x in markoutercomma(m.group('dims')).split('@,@')]
!                 n = rmbadname1(m.group('name').strip())
                  if n in block['vars']:
                      if 'attrspec' in block['vars'][n]:
                          block['vars'][n]['attrspec'].append(

```

**Comment by charris** [comment]
@pearu Comment?

**Comment by bobeldering** [comment]
Pull request: #9247

**Comment by pearu** [comment]
My comment: the issue is plausible and the pull request looks good to me.

**Comment by charris** [comment]
Closed by #9247.
